addons/base_vat/models/res_partner.py

```diff
@@ -750,6 +751,16 @@ def check_vat_br(self, vat):
         is_cnpj_valid = stdnum.get_cc_module('br', 'cnpj').is_valid
         return is_cpf_valid(vat) or is_cnpj_valid(vat)
 
+    __check_vat_cr_re = re.compile(r"^\d{9,12}$")
```

Comment:
This regex doesn't check for no leading zeros as described in the comments
`^[1-9]\d{8,11}$` maybe?

Reply:
EDIT: from what I see in the document NITE & JURÍDICA do allow leading zeros. In that case, the regex is ok, but the comment is not
or we could go more detailed
`^(?:[1-9]\d{8}|\d{10}|[1-9]\d{10}|[1-9]\d{11})$`

Reply:
@h4818 you have to prefix the regex pattern string with `r` to make it a raw string:

```python
 __check_vat_cr_re = re.compile(r'^(?:[1-9]\d{8}|\d{10}|[1-9]\d{10,11})$')
 ```
 
@luistorresm 
To avoid the W605 warning (which indicates invalid escape sequences in a string), you should use raw string literals for your regex patterns. Raw strings treat backslashes as literal characters and prevent them from being interpreted as escape characters.

Pura vida ! 🇨🇷

Reply:
@h4818 Hi! Hope you are OK.

Any news about this PR ?

Reply:
Hi @h4818 do you have any news about this PR?

Regards,

Reply:
Hi @randall-vx

The commit message and comment in the function are still not correct
The 10 digit code could contain leading 0's.
Once that's removed, I will merge it.

Reply:
What do you think now @h4818 ?

Reply:
I actually preferred the last regex. Only the comment was invalid.

But let's see how this one goes ;)